pkg/dependencies/factory.go

```diff
@@ -914,6 +921,33 @@ func (f *Factory) WithCuratedPackagesRegistry(registryName, kubeVersion string,
 	return f
 }
 
+func (f *Factory) WithRESTClient(kubeConfigFilename string) *Factory {
+	f.buildSteps = append(f.buildSteps, func(ctx context.Context) error {
+		loader := clientcmd.NewDefaultClientConfigLoadingRules()
+		if f.kubeConfigFilename != "" {
```

Comment:
What's the idea here? I got a bit lost

It seems like we can store the kubeconfig as part of the factory properties and then the rest client build logic uses it (if set) to configure the client. But we are also taking  `kubeConfigFilename` as an argument to `WithRESTClient`, except we seem to not use it, unless I'm missing something.

I think we can go with the simplest solution for now and just use the input argument to `WithRESTClient`? That way we avoid having to store config in the factory.

Reply:
Oof yes, this was a "should I do it this way?", no I should do it this other way... But then goofed.

My goal was to have the kubeconfig only specified in one place. But the trick is that it's optional in its previous existence, but withrestclient requires it, so yeah, it's probably best to specify it here, since it's required here.